apps/transport/lib/transport/vault.ex

```diff
@@ -22,8 +18,8 @@ defmodule Transport.Vault do
   defp key do
     case Application.fetch_env!(:transport, :app_env) do
       env when env in [:production, :staging] ->
-        @prod_config_env_name
-        |> System.fetch_env!()
+        :transport
+        |> Application.fetch_env!(:cloak_key)
```

Comment:
J’ai lu la description de la PR mais je n’ai pas compris comment la mise à jour de Sentry a  cassé le module qui va déchiffrer les tables chiffrées de notre base de données.

Reply:
C'était déjà cassé avant mais ce n'est pas visible. Exécute `mix test apps/transport/test/transport_web/controllers/page_controller_test.exs:135` sur `master` actuellement.

C'est lié à l'ordre de démarrage de l'application. Si le GenServer `Transport.Vault` est lancé avant le swap d'environnement, l'erreur ne va pas être visible.

Reply:
Trouvé, il y avait autre chose qui était requis par l'app en environnement `staging`. La partie importante est dans `CustomSecureBrowserHeaders`.

```
%{
 kind: :error,
 layout: {TransportWeb.LayoutView, "app.html"},
 reason: %ArgumentError{
   message: "could not fetch application environment :csp_url for application :sentry because configuration at :csp_url was not set"
 },
 stack: [
   {Application, :fetch_env!, 2, [file: ~c"lib/application.ex", line: 775]},
   {TransportWeb.Plugs.CustomSecureBrowserHeaders, :csp_headers, 2,
    [
      file: ~c"lib/transport_web/plugs/custom_secure_browser_headers.ex",
      line: 61
    ]},
   {TransportWeb.Plugs.CustomSecureBrowserHeaders, :call, 2,
    [
      file: ~c"lib/transport_web/plugs/custom_secure_browser_headers.ex",
      line: 12
    ]},
   {TransportWeb.Router, :browser, 2, []},
   {TransportWeb.Router, :__pipe_through1__, 1,
    [file: ~c"lib/transport_web/router.ex", line: 1]},
   {Phoenix.Router, :__call__, 2,
    [file: ~c"lib/phoenix/router.ex", line: 346]},
   {TransportWeb.Router, :call, 2,
    [file: ~c"lib/transport_web/router.ex", line: 1]},
   {TransportWeb.Plugs.Router, :"-dispatch/2-fun-0-", 4,
    [
      file: ~c"/Users/antoineaugusti/Documents/transport-site/deps/plug/lib/plug/router.ex",
      line: 246
    ]},
   {:telemetry, :span, 3,
    [
      file: ~c"/Users/antoineaugusti/Documents/transport-site/deps/telemetry/src/telemetry.erl",
      line: 321
    ]},
   {TransportWeb.Plugs.Router, :dispatch, 2,
    [
      file: ~c"/Users/antoineaugusti/Documents/transport-site/deps/plug/lib/plug/router.ex",
      line: 242
    ]},
   {TransportWeb.Plugs.Router, :plug_builder_call, 2,
    [file: ~c"lib/transport_web/plugs/router.ex", line: 1]},
   {TransportWeb.Endpoint, :plug_builder_call, 2,
    [file: ~c"lib/transport_web/endpoint.ex", line: 1]},
   {TransportWeb.Endpoint, :"call (overridable 3)", 2,
    [file: ~c"lib/transport_web/endpoint.ex", line: 1]},
   {TransportWeb.Endpoint, :call, 2,
    [file: ~c"lib/transport_web/endpoint.ex", line: 1]},
   {Phoenix.ConnTest, :dispatch, 5,
    [file: ~c"lib/phoenix/test/conn_test.ex", line: 225]},
   {TransportWeb.PageControllerTest,
    :"test robots.txt it works in staging with a different content", 1,
    [
      file: ~c"test/transport_web/controllers/page_controller_test.exs",
      line: 141
    ]},
   {ExUnit.Runner, :exec_test, 2,
    [file: ~c"lib/ex_unit/runner.ex", line: 463]},
   {:timer, :tc, 1, [file: ~c"timer.erl", line: 166]},
   {ExUnit.Runner, :"-spawn_test_monitor/4-fun-1-", 5,
    [file: ~c"lib/ex_unit/runner.ex", line: 385]}
 ],
 status: 500
}
```

`:sentry, :csp_url` était toujours défini auparavant dans tous les environnements, ce qui ne posait pas de problème.